tests/unit-core/test-regexp-dotall-unicode.c

```diff
@@ -0,0 +1,81 @@
+ /* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript.h"
+
+#include "test-common.h"
+
+int
+main (void)
+{
+  TEST_INIT ();
+  jerry_init (JERRY_INIT_EMPTY);
+
+  if (!jerry_is_feature_enabled (JERRY_FEATURE_SYMBOL))
+  {
+    jerry_port_log (JERRY_LOG_LEVEL_ERROR, "ES.next support is disabled\n");
+    jerry_cleanup ();
+    return 0;
+  }
+
+  jerry_value_t global_obj_val = jerry_get_global_object ();
+
+  jerry_char_t pattern2[] = "\\u{61}.\\u{62}";
+  uint16_t flags = JERRY_REGEXP_FLAG_DOTALL | JERRY_REGEXP_FLAG_UNICODE | JERRY_REGEXP_FLAG_STICKY;
+  jerry_value_t regex_obj = jerry_create_regexp (pattern2, flags);
+  TEST_ASSERT (jerry_value_is_object (regex_obj));
+
+  const jerry_char_t func_resource[] = "unknown";
+  const jerry_char_t func_arg_list[] = "regex";
+  const jerry_char_t func_src2[] = "return [regex.exec('a\\nb'), regex.dotAll, regex.sticky, regex.unicode ];";
+  jerry_value_t func_val = jerry_parse_function (func_resource,
+                                                 sizeof (func_resource) - 1,
+                                                 func_arg_list,
+                                                 sizeof (func_arg_list) - 1,
+                                                 func_src2,
+                                                 sizeof (func_src2) - 1,
+                                                 JERRY_PARSE_NO_OPTS);
+
+  jerry_value_t res = jerry_call_function (func_val, global_obj_val, &regex_obj, 1);
```

Comment:
Do we need the `global_obj_val` as the `this` argument for the function? If I understand it correctly the function does not use it.

Reply:
Yes, you're right. I remove the `global _object_value` from the test case